worlds/marioland2/options.py

```diff
@@ -0,0 +1,165 @@
+    display_name = "Mario's Castle Midway Bell"
+
+
+class Coinsanity(Toggle):
+    """Shuffles the singular coins found freestanding and in question mark blocks into the item pool, and adds location
+    checks made by obtaining a sufficient number of coins in particular levels within a single playthrough."""
+    default_name = "Coinsanity"
+
+
+class CoinsanityChecks(Range):
+    """Number of Coinsanity checks.
+    A higher number means more checks, and smaller coin amounts per coin item in the item pool.
+    If Accessibility is set to Locations, auto-scroll levels may have a lower maximum count, which may lead to this
+    value being limited."""
+    default_name = "Coinsanity Checks"
+    range_start = 31
+    range_end = 2599
+    default = 150
+
+
+class DifficultyMode(Choice):
+    """Play in normal or easy mode. You can also start in Normal Mode with an "upgrade" to Easy Mode in the item pool,
+    or start in Easy Mode with a Normal Mode "trap" in the item pool."""
+    display_name = "Difficulty Mode"
+    option_normal = 0
+    option_easy = 1
+    option_normal_to_easy = 2
+    option_easy_to_normal = 3
+    default = 0
+
+
+class ShufflePipeTraversal(Choice):
+    """Single: Shuffle a Pipe Traversal item into the item pool, which is required to enter any pipes.
+    Split: Shuffle 4 Pipe Traversal items, one required for entering pipes from each direction.
+    Note that being unable to enter pipes is very limiting and affects nearly half of all levels."""
+    display_name = "Shuffle Pipe Traversal"
+    option_off = 0
+    option_single = 1
+    option_split = 2
+    default = 0
+
+
+class RandomizeEnemies(Toggle):
+    """Randomize enemies throughout levels."""
+    display_name = "Randomize Enemies"
+
+
+class RandomizePlatforms(Toggle):
+    """Randomize platforms throughout levels."""
+    display_name = "Randomize Platforms"
+
+
+class AutoScrollChances(NamedRange):
+    """Chance per eligible level to be made into an auto scroll level. Can also set to Vanilla to leave them unchanged."""
+    display_name = "Auto Scroll Chance"
+    range_start = 0
+    range_end = 100
+    special_range_names = {"vanilla": -1, "none": 0, "all": 100}
+    default = -1
+
+
+class AutoScrollMode(Choice):
+    """Always: Any auto scroll levels will always auto-scroll.
+    Global Trap Item: Auto scroll levels will only auto-scroll after obtaining the Auto Scroll trap item.
+    Level Trap Items: As with Trap Item, but there is a separate trap item for each auto scroll level.
+    Global Cancel Item: Auto Scroll levels will stop auto-scrolling after obtaining the Auto Scroll Cancel item.
+    Level Cancel Items: As with Cancel Item, but there is a separate cancel item for each auto scroll level.
+    Chaos: Each level will randomly always auto scroll, have an Auto Scroll Trap, or have an Auto Scroll Cancel item.
+    The effects of Trap and Cancel items are permanent! If Accessibility is not set to Locations,
```

Comment:
This line will become outdated with #1481

Reply:
Changed